### ISSUE TITLE
Provide light theme

### ISSUE THREAD

**Issue by dharkness** [FredrikNoren/ungit#148]
The dark colors in `styles.css` look nice, but since I keep my entire display inverted the colors are glaringly bright. I stripped all colors from the CSS file, but that makes the page difficult to use.

Has anyone created a light theme already? If not, is someone up to the task? :)

**Comment by philprobinson84** [comment]
I'm happy to take this on, with two goals:
1. implement a "light" theme, which shall simply be an inversion of all colours in the current dark theme (preview below).
2. refactor theme styles to more easily support other colour-schemes in future.

@yebrahim - is your use-case similar to that described by @dharkness?

Here's a very early preview of the light theme:
![ungit-light-theme-preview](https://cloud.githubusercontent.com/assets/1550779/20271173/da3bdcf8-aa81-11e6-8a16-f6ecd8091412.png)

**Comment by codingtwinky** [comment]
Two things, 

1. @philprobinson84: I want to apologize in advance as I think CSS and LESS files are not as well organized as it can be.
2. Please let me know how I can help make this happened

**Comment by yebrahim** [comment]
@philprobinson84 not really, I'm thinking about ungit's integration into other projects. Particularly, I'm working on including it into [Datalab](https://github.com/googledatalab/datalab). I think at a minimum it should support light-on-dark as well as dark-on-light, and be open to supporting more themes as well, so I totally agree with your two goals.

I'd be happy to help with the theming part, I don't think that simply inverting the dark theme provides for a good light theme. How do you switch the theme though, is that already built into ungit? I took a quick look at the source code but couldn't find any easy way of switching other than changing the css file link in `index.html`.

**Comment by philprobinson84** [comment]
@yebrahim, @codingtwinky thanks for the replies and support - I should have something ready to test / evaluate within the next few days. Here's a quick update:

**Theme Config Option**
There's no theme switching support so far - this is something I will implement as part of this. As @FredrikNoren suggested, I've added a "theme" option to the config - this is a string which will correspond to a theme name.

**Refactoring CSS LESS Files**
I've been through all the LESS files for all components, switching out all hard-coded colors for variables. The existing variables.less has been split: `variables-common.less` contains variables common to all themes (there are only a couple). Visual variables are now stored in `variables-<theme name>.less` files - currently there are currently two: `variables-dark.less` (existing theme) and `variables-light.less` (existing theme, inverted). 

**Themes to be Added**
I intend to include three themes:

1. dark - the existing dark theme
2. inverse - inversion of the existing dark theme for @dharkness 's use case
3. light - a proper light theme rather than inversion - I primarily use ungit with Atom, and therefore my intention is to go for a visual match to Atom's One Light theme.

**Processing HTML**
I intend to use the `grunt-processhtml` package to switch the CSS file included in the index.html based on the theme name. I've got this working well, but I'm new to grunt so if there's a better / easier way please let me know!

To work with `grunt-processhtml` the `index.html` in public has been moved into a `htmltemplates` subfolder, with the `grunt-processhtml` package used to replace the CSS file depending on the theme variable specified in the config.

**Next Steps**
* Modify LESS files to include the correct `variables_<theme>.less` based on the config variable
* Test both dark and inverse themes
* Create a light theme
* Test again
* Submit PR

Thoughts, feedback, comments and criticisms all welcomed!

**Comment by yebrahim** [comment]
@philprobinson84 I'd argue for making the theme switching a js thing in the runtime, for example a drop down in some kind of a settings pane. I'm guessing that this is something that ungit will need anyway, so it's better to work on it now rather than using grunt-processhtml then dropping it later on. I'm happy to help with this if you provide the proper LESS files.

**Comment by codingtwinky** [comment]
Can we please have UI changes as a separate PR and separate issue? 

Once functionality of applying theme is done I would like to have more discussion about this later to reduce merge conflict with existing UI changes and have more thought out UI

**Comment by nixomose** [comment]
for those of us who gave up trying to keep up with the web insanity and now do back end work, but need to be able to use ungit, this suggestion was great:
@tuhoojabotti - Thanks! That combined with removing the body's background color is perfect.

and here's how you do it
edit  /usr/lib/node_modules/ungit/public/index.html

change the <body> line to
<body style="-webkit-filter: invert(100%)">

and then  edit /usr/lib/node_modules/ungit/public/css/styles.css
go to the "body" section around line 257
and change 
background-color: #252833;
to
/*  background-color: #252833; */

or just remove it, restart ungit and voila, it becomes readable in bright rooms.

**Comment by viniciusribeiro-hotmart** [comment]
I'd like to point out another rather simple use case which I'd have thought obvious:
**Bright environments.**
Normally, I'm all in for **Dark mode in every-f-iing-thing**.
For years I've been using various extensions for Chrome make every website dark.
However that changes if I have to work on a light environment.

For e.g., I'm working from a rather bright balcony, because of the sudden mass-work-from-home imposed by the COVID-19 outbreak. I find myself always changing everything to light themes during daylight hours, because otherwise it's really hard to see, because of reflexes on the screen.

(OT: BTW Dark Reader's been great for this -- I even disabled the native dark mode in some sites like StackOverflow, because I can easily switch my Dark/Light Theme globally way easier with the extension shortcut.)
